packages/meditrak-server/src/apiV2/import/importSurveyResponses/importSurveyResponses.js

```diff
@@ -82,15 +99,56 @@ export async function importSurveyResponses(req, res) {
       const [tabName, sheet] = surveySheets;
       const deletedResponseIds = new Set();
       const questionIds = [];
+      let survey;
 
       // extract response ids and set up update batcher
       const { maxColumnIndex, maxRowIndex } = getMaxRowColumnIndex(sheet);
       const minSurveyResponseIndex = INFO_COLUMN_HEADERS.length;
       const surveyResponseIds = [];
+      const isGeneratedIdByColumnIndex = [];
+      const existingResponseDataByColumnIndex = [];
+
+      const getExistingResponseData = async columnIndex => {
+        if (!(columnIndex in existingResponseDataByColumnIndex)) {
+          const criteria = {
+            entityId: entityCodeToId[getInfoForColumn(sheet, columnIndex, 'Entity Code')],
+            date: getDateStringForColumn(sheet, columnIndex),
+          };
+          const responseData = await findExistingResponseData(models, survey, criteria);
+          existingResponseDataByColumnIndex[columnIndex] = responseData;
+        }
+
+        return existingResponseDataByColumnIndex[columnIndex];
+      };
+
       for (let columnIndex = minSurveyResponseIndex; columnIndex <= maxColumnIndex; columnIndex++) {
         const columnHeader = getColumnHeader(sheet, columnIndex);
-        if (checkIsNewSurveyResponse(columnHeader)) {
+        const importMode = getImportMode(columnHeader);
+
+        // For import modes where an existing response id is not specified,
+        // we need to detect a survey and use it to look for existing responses
+        const shouldDetectSurvey = [
+          IMPORT_MODES.NEW,
+          IMPORT_MODES.UPDATE,
+          IMPORT_MODES.MERGE,
+        ].includes(importMode);
```

Comment:
If down the line we wanted to add a new import mode, I could see it being quite annoying to have to touch each of these `array.includes` and `importMode === IMPORT_MODES.X`

I would maybe prefer declaring these upfront perhaps something like:
```
const importBehaviours = {
  UPDATE: {
    shouldDetectSurvey: true,
    shouldGenerateId: false,
    shouldDeleteEmptyAnswer: true,
    etc...
  },
  etc...
}
```

Reply:
I love this suggestion @EMcQ-BES !! I will push some changes in the weekend.

Cheers,
Kostas a.k.a. "the Open Source community"

Reply:
May have beat you to it, open source community :) Feel free to make changes if you like